src/plugins/discover/public/application/components/chart/download_csv_button/download_csv_button.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo } from 'react';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { FormattedMessage } from '@osd/i18n/react';
+import Papa from 'papaparse';
+// @ts-ignore
+import { saveAs } from '@elastic/filesaver';
+import moment from 'moment';
+import { OpenSearchSearchHit } from '../../../doc_views/doc_views_types';
+import { IndexPattern, UI_SETTINGS } from '../../../../../../data/common';
+import { getServices } from '../../../../opensearch_dashboards_services';
+import { DiscoverRootState, useSelector } from '../../../utils/state_management';
+import { buildColumns } from '../../../utils/columns';
+import { DOC_HIDE_TIME_COLUMN_SETTING } from '../../../../../common';
+import { getLegacyDisplayedColumns } from '../../default_discover_table/helper';
+
+export interface DownloadCsvButtonProps {
+  indexPattern: IndexPattern;
+  rows: OpenSearchSearchHit[];
+}
+
+export const selectColumnNames = ({
+  indexPattern,
+  hideTimeColumn,
+  isShortDots,
+}: {
+  indexPattern: IndexPattern;
+  hideTimeColumn: boolean;
+  isShortDots: boolean;
+}) => (state: DiscoverRootState) => {
+  const stateColumns = state.discover.columns;
+  // check if state columns is not undefined, otherwise use buildColumns
+  const columns = buildColumns(stateColumns || []);
+
+  // Handle the case where all fields/columns are removed except the time-field one
+  const adjustedColumns =
+    columns.length === 1 && columns[0] === indexPattern.timeFieldName
+      ? [...columns, '_source']
+      : columns;
+
+  const displayedColumns = getLegacyDisplayedColumns(
+    adjustedColumns,
+    indexPattern,
+    hideTimeColumn,
+    isShortDots
+  );
+  return displayedColumns.map((column) => column.name);
+};
+
+const formatRowsForCsv = ({
+  displayedColumnNames,
+  indexPattern,
+}: {
+  displayedColumnNames: string[];
+  indexPattern: IndexPattern;
+}) => (row: OpenSearchSearchHit) => {
+  return displayedColumnNames.map((colName) => {
+    const fieldInfo = indexPattern.fields.getByName(colName);
+
+    if (typeof row === 'undefined') {
+      return '';
+    }
+
+    if (fieldInfo?.type === '_source') {
+      const formattedRow = indexPattern.formatHit(row, 'text');
+      return JSON.stringify(formattedRow);
+    }
+
+    const formattedValue = indexPattern.formatField(row, colName, 'text');
+
+    if (typeof formattedValue === 'undefined') {
+      return '';
+    }
+
+    return formattedValue;
+  });
+};
+
+export const DownloadCsvButton = ({ indexPattern, rows }: DownloadCsvButtonProps) => {
+  const { uiSettings } = getServices();
+
+  const [isShortDots, hideTimeColumn] = useMemo(() => {
```

Comment:
is this useMemo actually doing anything useful here? looks like we're just getting ui settings values - nothing expensive going on. also noticed we're inconsistent with how we access ui settings in this file - as below we access it directly still

Reply:
I don't think its doing anything useful :D , i'll get rid of.

FYI i did it because its from this code: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx#L63-L70